tree/treeplayer/inc/ROOT/TDataFrameInterface.hxx

```diff
@@ -1066,26 +1067,33 @@ protected:
       }
 
       {
-         std::unique_ptr<TFile> ofile(TFile::Open(filename.c_str(), "RECREATE"));
-         TTree t(treename.c_str(), treename.c_str());
+         auto df = GetDataFrameChecked();
+         unsigned int nSlots = df->GetNSlots();
+         TBufferMerger merger(filename.c_str(), "RECREATE");
+         std::vector<std::shared_ptr<TBufferMergerFile>> files(nSlots);
+         std::vector<std::shared_ptr<TTree>> trees(nSlots);
```

Comment:
The files need to be shared, because the server returns std::shared_ptr<TFile> and keeps a weak_ptr<TFile> to them to check they are expired when the server goes down. The TTrees can probably be unique_ptr, but there is not std::make_unique in C++11, only make_shared.